Update SDK to be compatible with US (default), CA and EU regions

GreyNewfie opened the pull request:
- Add region-specific base URLs for APIs
- Add tests for region-specific base URLs in NotificationAPI
- Bump package version to 1.0.0

GreyNewfie committed:
Add region-specific base URLs for NotificationAPI

GreyNewfie committed:
Add tests for region-specific base URLs in NotificationAPI

- Add tests for each region to check if the response is successful
- Update QueryLogs_ValidData_ReturnsHttpResponseMessage to use  consistent JSON serialization settings in QueryLogs test

GreyNewfie committed:
Bump package version to 1.0.0

GreyNewfie committed:
Add GitHub Actions workflow for pull request pipeline

- Configure .NET build, test, and code formatting checks
- Trigger workflow on pull requests to main branch

GreyNewfie committed:
Add Makefile and update README for development workflow

- Create Makefile with targets for restore, build, test, and code formatting
- Update README with detailed development setup and contribution guidelines
- Add manual command instructions for developers

GreyNewfie committed:
Normalize whitespace and formatting across model classes

This commit performs a consistent whitespace and formatting pass across multiple model classes in the NotificationApi.Server project. No functional changes were made, ensuring uniform code style and readability.

GreyNewfie committed:
Add EditorConfig for consistent code formatting

Configure EditorConfig with comprehensive settings for C# and other file types to enforce uniform code style across the project

GreyNewfie committed:
Remove trailing whitespace and normalize file formatting

Clean up source files by removing unnecessary whitespace and ensuring consistent formatting across project files

GreyNewfie committed:
Remove EditorConfig file

Remove additional .editorconfig file to avoid conflicts with the original in the Models folder

GreyNewfie committed:
Normalize indentation in NotificationApiServerTests

Adjust indentation for test data objects to maintain consistent code formatting across test methods. No functional changes were made to the test logic.

GreyNewfie committed:
Move EditorConfig to directory root to get consistent formatting locally and remotely

GreyNewfie committed:
Normalize indentation in NotificationApiServerTests

GreyNewfie committed:
Rename GitHub Actions workflow job to Pull Request Pipeline

Modify the pull request workflow job name to match what the GitHub check expects

mbasadi commented on src/NotificationApi.Server/NotificationApi.Server.csproj:
```diff
@@ -5,7 +5,7 @@
     <Nullable>enable</Nullable>
     <PackageId>NotificationAPI</PackageId>
-    <Version>0.4.0</Version>
+    <Version>1.0.0</Version>
```
vwesion should go 0.5.0
why: if your user upgrade from 0.4.0->0.5.0 the code will work without any change
Node had to change because it was a breaking change

GreyNewfie replied:
Got it! I'll make that change.

mbasadi commented:
I have not review tests yet

GreyNewfie committed:
Update package version to 0.5.0

Modify the project version in to reflect a minor update instead of a major update because there are no breaking changes
